### PR TITLE
TUSC-208 Display marketplace purchases table

### PR DESCRIPTION
**Pull request by BrooklynLCodes**
… hook added, Page and routing changes to reflect the marketplace purchases

## Summary by Sourcery

Add a new Marketplace Purchases page wired into routing that displays marketplace purchase data from the RHSM API, including an empty state when no purchases are available.

New Features:
- Introduce a Marketplace Purchases page that lists marketplace purchases with provider account, marketplace, and date added.
- Add a reusable table component for rendering marketplace purchases and a corresponding empty state view when none exist.
- Expose a useMarketplacePurchases hook to fetch paginated marketplace purchases data from the RHSM API.

Tests:
- Add unit tests for the marketplace purchases API hook, covering success, empty, pagination, and error states.
- Add tests for the Marketplace Purchases page to verify table rendering, empty state behavior, and informational popover content.
- Add tests for the Marketplace Purchases table component to validate row rendering, formatting, and accessibility roles.
- Add tests for the NoMarketplacePurchases component to verify empty state messaging and icon rendering.

### CODE REVIEW

File: src/Components/MarketPlacePurchases/__tests__/MarketplacePurchasesTable.test.tsx [hunk 1]
```diff
@@ -0,0 +1,73 @@
+import React from 'react';
+import { screen } from '@testing-library/react';
+import { renderWithRouter } from '../../../utils/testing/customRender';
+import { MarketplacePurchasesTable } from '../MarketplacePurchasesTable';
+import { MarketplacePurchase } from '../../../hooks/api/useMarketplacePurchases';
+
+jest.mock('../../../hooks/util/dates', () => ({
+  formatDate: (date: string) => `Formatted:${date}`,
+}));
```
Comment (lukegrn) [inline src/Components/MarketPlacePurchases/__tests__/MarketplacePurchasesTable.test.tsx]: Question: Why do we need to mock the date formatting function? Can it not just format the dates?

File: src/Components/MarketPlacePurchases/__tests__/MarketplacePurchasesTable.test.tsx [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    renderTable([]);
+
+    expect(
+      screen.getByRole('grid', {
+        name: /marketplace purchases table/i,
+      }),
+    ).toBeInTheDocument();
+
+    expect(screen.getAllByRole('row')).toHaveLength(1);
+  });
```
Comment (lukegrn) [inline src/Components/MarketPlacePurchases/__tests__/MarketplacePurchasesTable.test.tsx]: Cleanup: remove this test as it does not makes sense. The table will never render when there are no purchases as the empty state will

File: src/hooks/api/useMarketplacePurchases.ts [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+  const json = await response.json();
+
+  return json as MarketplacePurchasesResponse;
+};
+
+export const useMarketplacePurchases = (
+  args: FetchMarketplacePurchasesArgs,
+) => {
+  return useQuery({
+    queryKey: ['marketplacePurchases', args],
```
Comment (lukegrn) [inline src/hooks/api/useMarketplacePurchases.ts]: Question: Does passing an object as the second key here work? I have only ever seen these keyed with primitives. I know with effects in react an object _won't_ work as a key because it uses the pointer as the key value, which doesn't always change if the underlying data does.

May be worth testing and deciding is something like `['marketplacePurchases', args.limit, args.offset]` is better

File: src/Pages/MarketplacePurchasesPage/MarketplacePurchasesPage.tsx [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+  } = useMarketplacePurchases({
+    limit: 10,
+    offset: 0,
+  });
+
+  const marketplacePurchases = data?.body ?? [];
+  const hasMarketplacePurchases = marketplacePurchases.length > 0;
+
+  if (isMarketplacePurchasesLoading) return <Loading />;
+  if (isMarketplacePurchasesError) return <Unavailable />;
```
Comment (lukegrn) [inline src/Pages/MarketplacePurchasesPage/MarketplacePurchasesPage.tsx]: Issue: This page needs to check for user permissions and render no permissions if the user is missing those

File: src/Pages/MarketplacePurchasesPage/MarketplacePurchasesPage.tsx [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+
+const MarketplacePurchasesPage = () => {
+  const {
+    data,
+    isError: isMarketplacePurchasesError,
+    isLoading: isMarketplacePurchasesLoading,
+  } = useMarketplacePurchases({
+    limit: 10,
+    offset: 0,
+  });
```
Comment (lukegrn) [inline src/Pages/MarketplacePurchasesPage/MarketplacePurchasesPage.tsx]: Note: currently this calls with pagination parameters, with no way to update them, so we can only ever fetch the first 10. @BrooklynLCodes can you take an action item to make sure there is a ticket that covers the scope of adding pagination to this?

File: src/Components/MarketPlacePurchases/__tests__/MarketplacePurchasesTable.test.tsx (hunk 1 above)
Comment (BrooklynLCodes) [reply src/Components/MarketPlacePurchases/__tests__/MarketplacePurchasesTable.test.tsx]: I followed the existing pattern in CloudAccountsTable.test.tsx, which mocks formatDate so the table test can focus on rendering the returned value. I’m fine removing the mock and asserting the actual formatted date if that’s the preferred approach here.

File: src/hooks/api/useMarketplacePurchases.ts (hunk 3 above)
Comment (BrooklynLCodes) [reply src/hooks/api/useMarketplacePurchases.ts]: Good point. TanStack Query does support objects in query keys, but since I removed the pagination arguments from this hook, I simplified the query key to just ['marketplacePurchases'].

File: src/Pages/MarketplacePurchasesPage/MarketplacePurchasesPage.tsx (hunk 5 above)
Comment (BrooklynLCodes) [reply src/Pages/MarketplacePurchasesPage/MarketplacePurchasesPage.tsx]: I removed the remaining pagination references

Comment (lukegrn) [reply src/Pages/MarketplacePurchasesPage/MarketplacePurchasesPage.tsx]: Sure, either is fine.

Can you make sure there's a ticket that covers that scope?

**Review by lukegrn — APPROVED** [review]

**Review by kjmccauley — APPROVED** [review]
